### ISSUE TITLE
Supporting OCaml 4.14 with ocamlnat

### ISSUE THREAD

**Issue by aqjune** [dbuenzli/omod#15]
I wonder whether omod supports OCaml 4.14 with ocamlnat, or there is a plan to support it.

To enable ocamlnat from OCaml 4.14, I forked a repo and created a package: https://github.com/aqjune/opam-repo-ocamlnat/tree/master/packages/ocaml-variants.4.14.2%2Bocamlnat

In this setting, installing omod using `opam install omod` fails with this output message at `~/.opam/log/omod-....out`:

```
+ ocamlfind ocamlc -c -g -bin-annot -safe-string -package compiler-libs.toplevel -I src -I test -I src/support -o src/omod_nattop
.cmo src/omod_nattop.ml
findlib: [WARNING] Interface topdirs.cmi occurs in several directories: <snip>/_opam/lib/ocaml, <snip>/_opam/lib/ocaml/compiler-libs
File "src/omod_nattop.ml", line 7, characters 52-62:
7 |   Omod.Private.Top.set_topdirs ~is_nat:true (module Opttopdirs);
                                                        ^^^^^^^^^^
Error: Unbound module Opttopdirs
Command exited with code 2.
pkg.ml: [ERROR] cmd ['ocamlbuild' '-use-ocamlfind' '-classic-display' '-j' '4' '-tag' 'debug'
     '-build-dir' '_build' 'opam' 'pkg/META' 'CHANGES.md' 'LICENSE.md'
     'README.md' 'src/omod.a' 'src/omod.cmxs' 'src/omod.cmxa' 'src/omod.cma'
     'src/omod_top.cmx' 'src/omod.cmx' 'src/omod.cmi' 'src/omod.mli'
     'src/omod_nattop.a' 'src/omod_nattop.cmxs' 'src/omod_nattop.cmxa'
     'src/omod_nattop.cma' 'src/omod_nattop.cmx' 'src/omod.cmx'
     'src/omod.cmi' 'src/omod.mli' 'src/support/omod_support.a'
     'src/support/omod_support.cmxs' 'src/support/omod_support.cmxa'
     'src/support/omod_support.cma' 'src/support/omod_support.cmx'
     'src/support/omod_support.cmi' 'src/support/omod_support.mli'
     'src/support/omod_ocamlc.cmx' 'src/support/omod_ocamlc.cmi'
     'src/support/omod_ocamlc.mli' 'src/support/omod_cu.cmx'
     'src/support/omod_cu.cmi' 'src/support/omod_cu.mli' 'src/omod.top'
     'src/omod.nattop' 'src/omod_bin.native' 'doc/index.mld'
     'doc/tutorial.mld']: exited with 10
```

My understanding is that since OCaml 4.13 `Opttopdirs` is replaced with `Topdirs`.

**Comment by dbuenzli** [comment]
Perhaps. These things are a bit hard to follow since AFAIK they are not documented in the release notes and they are not tested by opam-ci. If they eventually unified the Topdirs it may actually simplify things.

**Comment by aqjune** [comment]
I could find it from a full log of Changes: https://github.com/ocaml/ocaml/blob/trunk/Changes#L3654-L3657 This seems to be changed since OCaml 4.13.
Seemingly relevant previous patches are: https://github.com/ocaml/ocaml/pull/10061 https://github.com/ocaml/ocaml/pull/10078 
If omod is experimentarily updated to use the unified toplevel modules, I can extensively run tests for my use cases and see how it goes.

**Comment by dbuenzli** [comment]
I pushed a quick fix to both `down` and `omod` you should be able to access them with:

```
opam pin add omod --dev
opam pin add down --dev
```
I need a bit more time to see how simpler all that can be now. Perhaps we can even move to remove the `omod.top`, `omod.nattop` dicotomy.

**Comment by aqjune** [comment]
Hi @dbuenzli , I managed to find a time to try omod. I started with installing a local switch of 4.14.2+ocamlnat from your repo (https://github.com/dbuenzli/opam-repo-ocamlnat), and then installed omod using the opam pin add --dev command.

I had two libraries I wanted to load using omod: zarith and camlp5.

Loading Zarith using omod was straightforward:
```
# #use "omod.top";;
Omod v0.0.4-3-gfa332ef loaded. Type Omod.help () for more info.
# Omod.load "Z";;
[INC] <..snip..>/_opam/lib/ocaml
[INC] <..snip..>/_opam/lib/zarith
[OBJ] <..snip..>/_opam/lib/zarith/zarith.cmxs
- : bool = true
# Omod.load "Q";;
- : bool = true
``` 

However, loading camlp5 (8.02.00) was not. Its main module is Pcaml, but loading it raised a following error:
```
# Omod.load "Pcaml";;
[INC] <..snip..>/_opam/lib/ocaml
[INC] <..snip..>/_opam/lib/ocaml/compiler-libs
[INC] <..snip..>/_opam/lib/camlp-streams
[OBJ] <..snip..>/_opam/lib/camlp-streams/camlp_streams.cmxs
[INC] <..snip..>/_opam/lib/camlp5
[OBJ] <..snip..>/_opam/lib/camlp5/odyl.cmxs
[ERROR] load <..snip..>/_opam/lib/camlp5/odyl.cmxs:
File not found: <..snip..>/_opam/lib/camlp5/odyl.cmxs

- : bool = false
#
```

I looked into `_opam/lib/camlp5` and indeed there was no `odyl.cmxs`, but only `odyl.cmx` and `odyl.cmxa`. How can I resolve this error?

Thanks again,

**Comment by dbuenzli** [comment]
That's likely an issue with `camlp5` which doesn't build and install `.cmxs` files.

**Comment by aqjune** [comment]
I appreciate the info!

**Comment by dbuenzli** [comment]
Try to open an issue https://github.com/camlp5/camlp5/issues and/or get in touch with @chetmurthy.
